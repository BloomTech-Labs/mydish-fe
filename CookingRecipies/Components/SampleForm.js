import React, { Component } from 'react';
import {
  Button,
  TextInput,
  View,
} from 'react-native';

export default class SampleForm extends Component {
  constructor(params) {
    super(params);
    this.state = { inputText: '' };
  }

  handleChangeText(text) {
    this.setState({ inputText: text });
  }

  handleSend() {
    const { inputText } = this.state;
    const { onSend } = this.props;

    if (onSend) {
      onSend(inputText);
    }

    this.setState({ inputText: '' });
  }

  render() {
    const { inputText } = this.state;
    return (
      <View>
        <TextInput
          value={inputText}
          testID="messageText"
          onChangeText={text => this.handleChangeText(text)}
        />
        <Button
          title="Send"
          testID="sendButton"
          onPress={() => this.handleSend()}
        />
      </View>
    );
  }
}