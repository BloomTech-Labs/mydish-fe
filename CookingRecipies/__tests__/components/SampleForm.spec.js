import React from 'react';
import {render, fireEvent} from 'react-native-testing-library';
import SampleForm from '../../Components/SampleForm';
import renderer from 'react-test-renderer';

describe('SampleForm', () => {
    it('clears the message field', () => {
        const {getByTestId} = render(<SampleForm/>);
        const TextInput = getByTestId('messageText');
        fireEvent.changeText(TextInput, 'Hello World');
        expect(TextInput.props.value).toEqual('Hello World');
        const Button = getByTestId('sendButton');
        fireEvent.press(Button);
        expect(TextInput.props.value).toEqual('');
    });

    it('SampleForm Snapshot', () => {
        const SFSnap = renderer.create(<SampleForm/>).toJSON();
        expect(SFSnap).toMatchSnapshot();
    })

    it('Send handler triggered upon Button press', () => {
        const messageTxt = 'Hello World';
        const sendHandler = jest.fn();
        const SampleForm = render(<SampleForm onSend={sendHandler} />);

        const TxtInput = SampleForm.getByTestId('messageText');

        fireEvent.changeText(TxtInput, messageTxt);
        fireEvent.press(TxtInput);
        expect(sendHandler).toHaveBeenCalledWith(messageTxt);
    });

    





});


