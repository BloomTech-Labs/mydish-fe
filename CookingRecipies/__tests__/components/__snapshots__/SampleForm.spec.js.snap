// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleForm SampleForm Snapshot 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    testID="messageText"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="sendButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Send
      </Text>
    </View>
  </View>
</View>
`;
